### PR TITLE
fix(release): ensure component package.json file has correct data

### PR DESCRIPTION
**Pull request by Hyperkid123**
cc @dlabaj 

### Changes
- fixes up the build for the module federation environment
  - create a new `dynamic` directory in the `dist` folder for all dynamically resolved modules
  - enable dynamic output for the icons package
  - remove duplicate runs for different output types (ems, cjs)

Based on local testing the packages are correctly shared and "tree shaken" if we use the absolute module paths to the `dynamic` directory.

> __NOTE__: I am using the `pf-5` alias for the module because I also have the v4 running in my test env

**Import example**

```jsx
import { Button as PF5Button } from 'pf-5/dist/dynamic/components/Button';
import { Icon } from 'pf-5/dist/dynamic/components/Icon';
import AdIcon from 'pf-5-icons/dist/dynamic/icons/ad-icon';
```
**Module federation sharing config example**

```js
    new ModuleFederationPlugin({
    ...,
      shared: [
            {
              'pf-5/dist/dynamic/components/Button': {
                requiredVersion: '5.0.0-prerelease.8',
              },
              'pf-5-icons/dist/dynamic/icons/ad-icon': {
                requiredVersion: '5.0.0-prerelease.3',
              },
           }
        ],
    })
```

**Shared scope result**

the `import { Icon } from 'pf-5/dist/dynamic/components/Icon';` was not marked for sharing deliberately, hence it's not in the shared scope.

```js
{
  __webpack_share_scopes__: {
    default: {
      'pf-5-icons/dist/dynamic/icons/ad-icon': {
        '5.0.0-prerelease.3': { from: 'landing-page-frontend', eager: false, get: ƒ },
      },
      'pf-5/dist/dynamic/components/Button': {
        '5.0.0-prerelease.8': { from: 'landing-page-frontend', eager: false, get: ƒ },
      },
    },
  },
}
```

Only these modules were used in the testing and only these modules were included within the webpack build. Thanks to the absolute import paths. All shared modules were loaded only once during runtime

### CODE REVIEW

File: packages/react-table/single-packages.config.json [hunk 1]
```diff
@@ -2,16 +2,8 @@
     "packageName": "@patternfly/react-table",
     "modules": [ 
         {
-            "name": "components",
+            "name": "dist/esm",
             "type": "esm"
-        },
-        {
-            "name": "deprecated",
```
Comment (nicolethoen) [inline packages/react-table/single-packages.config.json:9]: Are deprecated components are still being delivered via these federated modules? They will need to be

Comment (Hyperkid123) [reply packages/react-table/single-packages.config.json:9]: I have a call with @dlabaj where I want to clear up some questions. The deprecated modules is one of them.

Comment (dlabaj) [reply packages/react-table/single-packages.config.json:9]: They are the build script has been updated to point to the deprecated components.  In the dynamic folder it contains all the esm versions for the deprecated components.

File: packages/react-charts/single-packages.config.json [hunk 2]
```diff
@@ -4,10 +4,6 @@
         {
             "name": "dist/esm",
             "type": "esm"
-        },
-        {
```
Comment (dlabaj) [inline packages/react-charts/single-packages.config.json:8]: Do we not want common js anymore?

Comment (dlabaj) [reply packages/react-charts/single-packages.config.json:8]: okay I see they are in the package.json under main.

File: scripts/build-single-packages.js [hunk 3]
```diff
@@ -19,27 +20,42 @@ if (!configJson.modules || configJson.modules.length === 0) {
...
   const packageName = configJson.packageName;
   if (!packageName) {
     console.log("packageName is required!")
     process.exit(1);
   }
 
-  let componentName = pathAsArray[pathAsArray.length - 2];
+  let componentName = pathAsArray[pathAsArray.length - (component.match(/index\.js$/) ? 2 : 1)];
   if (pathAsArray.includes("next")) {
     componentName = `${componentName.toLowerCase()}-next-${dist}`;
```
Comment (dlabaj) [inline scripts/build-single-packages.js:60]: Thinking we should change this to dynamaic instead of the dist type since we only have one package.json with a main and a module in it.  thoughts?

File: packages/react-charts/single-packages.config.json (hunk 2 above)
Comment (Hyperkid123) [reply packages/react-charts/single-packages.config.json:8]: We do, but we don't have to generate `package.json` per each building type. We only need one that references the others. I think we might even consider scanning the src folder instead of the output.

File: packages/react-table/single-packages.config.json (hunk 1 above)
Comment (Hyperkid123) [reply packages/react-table/single-packages.config.json:9]: Yeah, we no longer need separate builds. All of the files are emitted during a single cycle now.

**Comment by Hyperkid123** [comment]
@dlabaj I've cleaned up the script. I've also found all the potential useful transform import plugins for various build tools. I'll continue with more testing in HCC, but it would be very useful if we can get another pre-release so that I don't have to test with locally linked packages. There might be some edge cases that I won't be able to cover with locally linked packages.

**Review by dlabaj — APPROVED** [review]
LGTM

**Review by tlabaj — APPROVED** [review]
